### PR TITLE
Add --upload_rpms option.  Need this so we can use pbench on betas in…

### PR DESCRIPTION
**Pull request by dvalinrh**
… the cloud

### CODE REVIEW

File: ansible_roles/roles/install_pbench/tasks/main.yml [hunk 1]
```diff
@@ -24,6 +24,12 @@
       delegate_to: "{{ target_sys }}"
       become: yes
 
+- name: Upload and stall any rpms.
```
Comment (kdvalin) [inline ansible_roles/roles/install_pbench/tasks/main.yml]:
```suggestion
- name: Upload and install any rpms.
```

File: bin/ten_of_us.yml [hunk 2]
```diff
@@ -424,6 +437,21 @@
...
 
+- hosts: install_group
+  user: config_info.test_user
+  become_user: root
+  become: true
+  vars_files:
+    - ansible_vars.yml
+  gather_facts: yes
+  tasks:
+  - name: Upload and stall any rpms.
```
Comment (kdvalin) [inline bin/ten_of_us.yml]:
```suggestion
  - name: Upload and install any rpms.
```

**Review by kdvalin — CHANGES_REQUESTED** [review]
Minor typo fixes needed

**Comment by dvalinrh** [comment]
Pushed set of changes
1) Fixes it so we install the rpms on all systems when network is designated
2) Fixes issue where --insall_rpms was causing us to bail out of the scenario early even though there was more to do.
3) option to use spot now so it can be used within the scenario file as it should be.

**Review by kdvalin — APPROVED** [review]
LGTM

**Review by kdvalin — CHANGES_REQUESTED** [review]
GitHub did a weird thing, previous feedback still stands.

Minor typos

File: bin/burden [hunk 3]
```diff
@@ -2389,6 +2403,14 @@ convert_scenario_file()
 	if [ $? -eq 1 ]; then
 		cleanup_and_exit "The file, parse_file does not meet yaml requirements." 1
 	fi
+	#
```
Comment (frival) [inline bin/burden]: Fair warning: I'm going to start NAKing changes that include unrelated fixes because it makes finding and reverting breakages difficult.

File: bin/burden [hunk 4]
```diff
@@ -4063,7 +4096,7 @@ for i in `ls -d *`; do
 			date=`date`
 			echo "$data: $date tests completed: ${tests}, tests left ${tests_left}" >> /home/zathras_log/spot_fails
 		fi
-		sed "s/--use_spot/--no_spot_recover/g" $gl_top_dir/$file_to_use | sed "s/${tests}//g" > $gl_top_dir/rerun
+		sed "s/--use_spot 1/--no_spot_recover/g" $gl_top_dir/$file_to_use | sed "s/${tests}//g" > $gl_top_dir/rerun
```
Comment (frival) [inline bin/burden]: It looks like you're making "--use_spot 1" as an alias for "--no_spot_recover".  Why not just use "--no_spot_recover" where it's needed or if you have to make "--use_spot" take args at least make them symbolic rather than magic numbers?

File: bin/ten_of_us.yml [hunk 5]
```diff
@@ -319,6 +319,19 @@
       exit_msg: "Code repo installation failure"
     when: (config_info.os_vendor == "rhel") and (config_info.system_type == "aws" or config_info.system_type == "azure" or config_info.system_type == "gcp")
 
+ #
+ # Determine where to upload from, assumption is all test systems have the same
```
Comment (frival) [inline bin/ten_of_us.yml:323]: Ouch that's a heck of an assumption.

**Review by frival — CHANGES_REQUESTED** [review]
One is a future warning, another I'd like either an explanation for or a change to.  Fair 'nuff?

File: bin/ten_of_us.yml (hunk 2 above)
Comment (kdvalin) [reply bin/ten_of_us.yml]: This one's still here :)
